### PR TITLE
Use SingletonTrait in PermissionManager

### PR DESCRIPTION
**Pull request by Lunarelly**

### CODE REVIEW

**Comment by IvanCraft623** [comment]
This should be targeted to `minor-next`

**Comment by Lunarelly** [comment]
then merge it?

**Comment by dktapps** [comment]
I'm unsure if this is desirable, since SingletonTrait also allows overwriting and resetting the singleton. Not really sure if that makes sense for PermissionManager.

**Comment by Lunarelly** [comment]
maybe implement something like ReadonlySingletonTrait

File: src/permission/PermissionManager.php [hunk 1]
```diff
@@ -23,19 +23,12 @@
...
-	private static ?self $instance = null;
-
-	public static function getInstance() : PermissionManager{
-		if(self::$instance === null){
-			self::$instance = new self();
-		}
-
-		return self::$instance;
-	}
+	use SingletonTrait;
```
Comment (xxAROX) [inline src/permission/PermissionManager.php:31]:
```php
use SingletonTrait{
   reset as private;
   setInstance as private;
}
```

**Comment by dktapps** [comment]
Closing due to lack of activity.

**Pull request closed without merging**